consensus/src/quorum_store/proof_manager.rs

```diff
@@ -88,23 +128,137 @@ impl ProofManager {
                     PayloadFilter::InQuorumStore(proofs) => proofs,
                 };
 
-                let proof_block = self.proofs_for_consensus.pull_proofs(
-                    &excluded_batches,
-                    max_txns,
-                    max_bytes,
-                    return_non_full,
+                let (proof_block, proof_queue_not_fully_utilized) = self
+                    .proofs_for_consensus
+                    .pull_proofs(&excluded_batches, max_txns, max_bytes, return_non_full);
+
+                for excluded_batch in &excluded_batches {
```

Comment:
I don't see the need to do all of this state cleanup here (from this line, down to removing expired batches). To make things easier to understand, can we be consistent with the proofs and remove on commit and expiration? (The only difference AFAICT should be removing on getting the full proof)

Reply:
Earlier, the latency for 3000 TPS and 5000 TPS sweep tests was very high because we didn't cleanup these excluded batches. Cleaning up the excluded batches fixed the issue. These excluded batches won't be many. So, cleanup shouldn't take a lot of time hopefully.

Reply:
We have to filter them, but we don't have to remove them

Reply:
Done.